optimum/onnxruntime/modeling_diffusion.py

```diff
@@ -0,0 +1,423 @@
+        **kwargs,
+    ):
+        save_directory = Path(save_directory)
+        src_to_dst_path = {
+            self.vae_decoder_model_path: save_directory
+            / DIFFUSION_MODEL_VAE_DECODER_SUBFOLDER
+            / vae_decoder_file_name,
+            self.text_encoder_model_path: save_directory
+            / DIFFUSION_MODEL_TEXT_ENCODER_SUBFOLDER
+            / text_encoder_file_name,
+            self.unet_model_path: save_directory / DIFFUSION_MODEL_UNET_SUBFOLDER / unet_file_name,
+        }
+
+        # TODO: Modify _get_external_data_paths to give dictionnary
+        src_paths = list(src_to_dst_path.keys())
+        dst_paths = list(src_to_dst_path.values())
+        # Add external data paths in case of large models
+        src_paths, dst_paths = _get_external_data_paths(src_paths, dst_paths)
+
+        for src_path, dst_path in zip(src_paths, dst_paths):
+            dst_path.parent.mkdir(parents=True, exist_ok=True)
+            shutil.copyfile(src_path, dst_path)
+
+        self.tokenizer.save_pretrained(save_directory.joinpath("tokenizer"))
+        self.scheduler.save_pretrained(save_directory.joinpath("scheduler"))
+        if self.feature_extractor is not None:
+            self.feature_extractor.save_pretrained(save_directory.joinpath("feature_extractor"))
+
+    @classmethod
+    def _from_pretrained(
+        cls,
+        model_id: Union[str, Path],
+        config: Dict[str, Any],
+        use_auth_token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        cache_dir: Optional[str] = None,
+        vae_decoder_file_name: str = ONNX_WEIGHTS_NAME,
+        text_encoder_file_name: str = ONNX_WEIGHTS_NAME,
+        unet_file_name: str = ONNX_WEIGHTS_NAME,
+        local_files_only: bool = False,
+        provider: str = "CPUExecutionProvider",
+        session_options: Optional[ort.SessionOptions] = None,
+        provider_options: Optional[Dict[str, Any]] = None,
+        use_io_binding: Optional[bool] = None,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        **kwargs,
+    ):
+        if provider == "TensorrtExecutionProvider":
+            raise ValueError("The provider `'TensorrtExecutionProvider'` is not supported")
+
+        model_id = str(model_id)
+        sub_models_to_load, _, _ = cls.extract_init_dict(config)
+        sub_models_names = set(sub_models_to_load.keys()).intersection({"feature_extractor", "tokenizer", "scheduler"})
+        sub_models = {}
+
+        if not os.path.isdir(model_id):
+            allow_patterns = [os.path.join(k, "*") for k in config.keys() if not k.startswith("_")]
+            allow_patterns += list(
+                {
+                    vae_decoder_file_name,
+                    text_encoder_file_name,
+                    unet_file_name,
+                    SCHEDULER_CONFIG_NAME,
+                    CONFIG_NAME,
+                    cls.config_name,
+                }
+            )
```

Comment:
Doing set -> list because you want only unique values?

Reply:
Yes